src/docker.ts

```diff
@@ -37,24 +40,53 @@ const saveDockerImages = async (): Promise<void> => {
         "read-only option was selected."
     );
   } else {
-    const preexistingImages = getState(DOCKER_IMAGES_LIST).split("\n");
-    info("Listing Docker images.");
-    const images = await execBashCommand(LIST_COMMAND);
-    const imagesList = images.split("\n");
-    const newImages = imagesList.filter(
-      (image: string): boolean => !preexistingImages.includes(image)
-    );
-    if (newImages.length === 0) {
-      info("No Docker images to save");
+    /* Access the private method, getCacheEntry, internal to the GitHub Actions
+     * cache action.
+     */
+    interface ArtifactCacheEntry {
+      cacheKey?: string;
+      scope?: string;
+      cacheVersion?: string;
+      creationTime?: string;
+      archiveLocation?: string;
+    }
+    interface CacheWithPrivateMethod extends CacheType {
+      getCacheEntry(
+        keys: string[],
+        paths: string[]
+      ): Promise<ArtifactCacheEntry | null>;
+    }
```

Comment:
Nit: prefer declaring `interfaces` at the top level. They are a static concept handled by the compiler, so it's helpful to declare them statically.